[TACHYON-1346] optionally wait for process to finish/die

abridgett opened the pull request:
this is useful when running under cluster managers etc.

This PR superceeds #2159

jsimsa commented on integration/bin/tachyon-mesos.sh:
```diff
@@ -7,7 +7,37 @@ SCRIPT_DIR="$(cd "$(dirname "$0")"; pwd)"
+shift $((OPTIND-1))
+
 MESOS_MASTER_ADDRESS="$1"
+if [ ! "$MESOS_MASTER_ADDRESS" ]; then
```
[[ ]]

jsimsa replied:
@{MESOS_MASTER_ADDRESS}

jsimsa commented on integration/bin/tachyon-mesos.sh:
```diff
@@ -7,7 +7,37 @@ SCRIPT_DIR="$(cd "$(dirname "$0")"; pwd)"
+
 MESOS_MASTER_ADDRESS="$1"
+if [ ! "$MESOS_MASTER_ADDRESS" ]; then
+  echo -e "$Usage"
```
${Usage}

jsimsa commented on integration/bin/tachyon-mesos.sh:
```diff
@@ -7,7 +7,37 @@ SCRIPT_DIR="$(cd "$(dirname "$0")"; pwd)"
+Usage="Usage: tachyon-mesos.sh [-hw] MESOS_MASTER_ADDRESS
++MESOS_MASTER_ADDRESS is of the form 'mesos.example.com:5050'
++
++-w  Wait for process to finish before returning
```
s/Wait/wait

jsimsa commented on bin/tachyon-start.sh:
```diff
@@ -28,6 +28,8 @@ MOPT is one of:
 
 -N  Do not try to kill prior running masters and/or workers in all or local
 
+-w  Wait for processes to end before returning
```
s/Wait/wait

apc999 replied:
I think -N has comment "Do not ...", starting with capitalized "D", so may be fine

jsimsa commented:
@abridgett thanks for the PR ... I left some minor comments ... overall this LGTM

@apc999 please take a quick look

AmplabJenkins commented:
Test PASSed.
Refer to this link for build results (access rights to CI server needed): 
https://amplab.cs.berkeley.edu/jenkins//job/Tachyon-Pull-Request-Builder/7355/
Test PASSed.

apc999 commented:
+1 LGTM

abridgett commented:
I've fixed the syntax (sorry!)
Corrected the case (also the line above which I'd based it from).    I think it's a little more readable capitalised so if you like I'll change them all.

apc999 commented:
@abridgett Thanks. The PR looks good now.

I will merge it

jsimsa commented:
LGTM

AmplabJenkins commented:
Test PASSed.
Refer to this link for build results (access rights to CI server needed): 
https://amplab.cs.berkeley.edu/jenkins//job/Tachyon-Pull-Request-Builder/7356/
Test PASSed.
